windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1138 @@
+try
+{
+    size_t pathLength{ filepath.length() };
+
+    if (pathLength <= 0) {
+        promise.Reject("Invalid path.");
+    }
+    else {
+        bool hasTrailingSlash{ filepath[pathLength - 1] == '\\' || filepath[pathLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, pathLength - 1) : filepath);
+        path.make_preferred();
+
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(path.parent_path().wstring()) };
+        auto target{ co_await folder.GetItemAsync(path.filename().wstring()) };
+        co_await target.DeleteAsync();
+
+        promise.Resolve();
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) // FileNotFoundException
+    {
+        promise.Reject(RN::ReactError{ "ENOENT", "ENOENT: no such file or directory, open " + filepath });
+    }
+    else
+    {
+        // "Failed to unlink file" 
+        promise.Reject( winrt::to_string(ex.message()).c_str() );
+    }
+}
+
+
+winrt::fire_and_forget RNFSManager::exists(std::string filepath, RN::ReactPromise<bool> promise) noexcept
+try
+{
+    size_t fileLength{ filepath.length() };
+
+    if (fileLength <= 0) {
+        promise.Resolve(false);
+    }
+    else {
+        bool hasTrailingSlash{ filepath[fileLength - 1] == '\\' || filepath[fileLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, fileLength - 1) : filepath);
+
+        winrt::hstring directoryPath, fileName;
+        splitPath(filepath, directoryPath, fileName);
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+        if (fileName.size() > 0) {
+            co_await folder.GetItemAsync(fileName);
+        }
+        promise.Resolve(true);
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) {
+        promise.Resolve(false);
+    }
+    // "Failed to check if file or directory exists.
+    promise.Reject(winrt::to_string(ex.message()).c_str());
+}
+
+
+void RNFSManager::stopDownload(int32_t jobID) noexcept
+{
+    m_tasks.Cancel(jobID);
+}
+
+
+void RNFSManager::stopUpload(int32_t jobID) noexcept
+{
+    m_tasks.Cancel(jobID);
+}
+
+
+winrt::fire_and_forget RNFSManager::readFile(std::string filepath, RN::ReactPromise<std::string> promise) noexcept
+try
+{
+    winrt::hstring directoryPath, fileName;
+    splitPath(filepath, directoryPath, fileName);
+
+    StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+    StorageFile file{ co_await folder.GetFileAsync(fileName) };
+
+    Streams::IBuffer buffer{ co_await FileIO::ReadBufferAsync(file) };
+    winrt::hstring base64Content{ Cryptography::CryptographicBuffer::EncodeToBase64String(buffer) };
+    promise.Resolve(winrt::to_string(base64Content));
+}
+catch (const hresult_error& ex)
```

Comment:
This catch is also not reached if the file path points to non-existent file. These are the type of exceptions in the output:
```
Exception thrown at 0x00007FFED89D4B89 (KernelBase.dll) in MyApp.exe: WinRT originate error - 0x80070002 : 'The system cannot find the file specified.'.
Exception thrown at 0x00007FFED89D4B89 in MyApp.exe: Microsoft C++ exception: winrt::hresult_error at memory location 0x0000003BE39FD5F0.
```

Reply:
I noticed yesterday also these `The system cannot find the file specified` errors in my tests.

Reply:
But when i debug code it seems that it goes to catch and handles it. There is still that error in console `Exception thrown at 0x00007FFED89D4B89 (KernelBase.dll) in MyApp.exe: WinRT originate error - 0x80070002 : 'The system cannot find the file specified.'` maybe it is ok?